### PR TITLE
Fix bug 1699455: suppress warning when the dotenv file is missing

### PR DESCRIPTION
**Pull request by flozz**
With this PR, `manage.py` call the `dotenv.read_dotenv()` function only

* if the `DOTENV_PATH` environment variable is defined,
* if the `.env` or `.env/.env` file is exists (both are checked by the `dotenv` module).

This will suppress useless warnings.

Related bug → https://bugzilla.mozilla.org/show_bug.cgi?id=1699455

### CODE REVIEW

**Review by mathjazz — COMMENTED** [review]
Huh, this one fell through the cracks... Thanks for the patch, I tested on stage and locally and it works as expected.

There's also `pontoon/base/celeryapp.py`, which triggers a similar warning on deploy:
```
remote: /tmp/build_21ad6c66/pontoon/base/celeryapp.py:20: UserWarning: Not reading /tmp/build_21ad6c66/.env - it doesn't exist.
remote:   dotenv.read_dotenv(path(".env"))
```

Could you please update that one as well?

**Comment by flozz** [comment]
* I removed the warning in the celery app
* The celery app will now handle the `DOTENV_PATH` environment variable like `manage.py` and the WSGI app
* The WSGI app will now search for `.env` or `.env/.env` file at the Pontoon's root folder, like `manage.py` and the celery app

File: manage.py [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
-    warnings.filterwarnings('ignore', module='dotenv')
+if __name__ == "__main__":
+    # Read dotenv file and inject it's values into the environment
+    root_path = os.path.dirname(__file__)
+    if (
+        "DOTENV_PATH" in os.environ
+        or os.path.isfile(os.path.join(root_path, ".env"))
+        or os.path.isfile(os.path.join(root_path, ".env", ".env"))
+    ):
+        dotenv.read_dotenv(os.environ.get("DOTENV_PATH"))
```
[the remark this answers is not in the document]
Comment (flozz) [reply manage.py:16]: Both does the same as the dotenv library search for a `.env` or `.env/.env` file located in the same folder than the python file that call it... But you are right, it is probably better to have the same code everywhere :)

**Comment by mathjazz** [comment]
Sorry for throwing this in late, but it seems like the more elegant solution is to switch to https://saurabh-kumar.com/python-dotenv/, and then we don't need to bypass the warnings.

The only relevant change in the API seems to be `load_dotenv` instead of `read_dotenv`.

**Comment by flozz** [comment]
I replaced `django-dotenv` by `python-dotenv`.

* There is a breaking change: a `.env` file located in a `.env` folder at the root of the project (`.env/.env`) will not be read anymore
* There is a behaviour change too: `python-dotenv` will try to find a `.env` files in each parent folders until it finds one or it reaches the FS root.

**Review by mathjazz — COMMENTED** [review]
Thanks for the update, works great!

Please update the reference to `django-dotenv` in the docs and we're good.

**Review by mathjazz — APPROVED** [review]

**Pull request merged**